python/TestHarness/schedulers/QueueManager.py

```diff
@@ -504,63 +303,254 @@ def augmentQueueParamsBase(self, job_container):
-            # no longer do we need to nerf the TestHarness's os.walk routine...)
-            self.queueJobs(status_jobs=self.__jobs)
+        if os.path.exists(stdout_file):
+            file = open(stdout_file, 'r+')
+            outfile = file.read()
+        else:
+            self.adjustJobStatus(job, tester.bucket_fail, 'NO STDOUT FILE')
+            return ''
 
-        # Write session data to file
-        self.writeSessionFile()
+        # Alter test_dir to reflect working_dir creation
+        tester.specs['test_dir'] = json_job['working_dir']
+
+        if tester.hasRedirectedOutput(self.options):
+            outfile += util.getOutputFromFiles(tester, self.options)
+
+        # Allow the tester to verify its own output.
+        output = tester.processResults(tester.specs['moose_dir'], self.options, outfile)
+
+        # Set the testers output with modifications made above
+        job.setOutput(output)
+
+        # Append processResults to our stdout file
+        file.seek(0)
+        file.write(output)
+        file.truncate()
+        file.close()
 
-    def queueJobs(self, status_jobs=[], run_jobs=[]):
+        return output
+
+    def executeAndGetJobOutput(self, job):
+        """
+        Execute derived commands and obtain the output
         """
-        Entry point into the QueueManager. Supply queueJobs with a list
-        of jobs to either launch, or check statuses for.
+        json_data = self.getData(job.getUniqueIdentifier(), working_dir=True)
+        output = util.runCommand(self.getQueueCommand(job), cwd=json_data['working_dir'])
+        return output
 
-        Syntax:
+    def _adjustDownstreamTests(self, job):
+        """
+        adjust downstream tests to queued to prevent them
+        from being skipped
+        """
+    def getCurrentJobStatus(self, job):
+        """
+        Set a job to the most current known status.
+
+        This method will first check if job has a previous pass/fail
+        status. If so, it sets that status with out performing any
+        third party queueing commands.
+
+        If the job has a status of queued, we call derived methods to
+        determine current status as the third party queueing system
+        sees it.
+        """
+        tester = job.getTester()
+        json_status = self.getData(job.getUniqueIdentifier(),
+                                   status_bucket=True,
+                                   caveat_message=True,
+                                   std_out=True,
+                                   working_dir=True)
+        if json_status:
+            bucket = self.__createStatusBucket(json_status)
+
+        # This test was not included during the launch sequence (was silently skipped)
+        else:
+            self.adjustJobStatus(job, tester.bucket_silent, 'NO LAUNCH INFORMATION')
+            return
+
+        self.adjustJobStatus(job, bucket, json_status['caveat_message'])
+
+        third_party_job_information = {}
```

Comment:
third party? You mean the PBS here right? I'm not sure this is the best name of the variable but it's probably ok